system/Config/Factories.php

```diff
@@ -422,7 +423,7 @@ public static function setOptions(string $component, array $values): array
      */
     public static function reset(?string $component = null)
     {
-        if ($component) {
+        if ($component !== null && $component !== '') {
```

Comment:
```suggestion
        if ((string) $component !== '') {
```

Reply:
Only `$component !== null` is sufficient.

Reply:
updated to ony check `!==null`